manifests/init.pp

```diff
@@ -16,29 +16,49 @@
   $resolver_config_file_owner  = 'root',
   $resolver_config_file_group  = 'root',
   $resolver_config_file_mode   = '0644',
-) {
+  ){
 
-  # Validates domain
-  if is_domain_name($domain) != true {
-    fail("Domain name, ${domain}, is invalid.")
+  if $::kernel == 'windows' {
+    if is_array($nameservers) {
+      $stringifynameservers = join($nameservers,',')
+      $testserver = $nameservers[0]
+    }
+    else {
+      $stringifynameservers = $nameservers
+      $testserver = $nameservers
+    }
+    exec { 'set dns':
+      command  => "\$interface = find-netroute -RemoteIPAddress ${testserver} | select -expand 'InterfaceAlias'; set-dnsclientserveraddress -interfacealias \$interface -serveraddress {${stringifynameservers}}",
```

Comment:
I could replace the command and unless with templates of powershell scripts with a for loop to allow me to handle dns entries on multiple interfaces, and also make this much more readable.  I'll take a whack at that tomorrow.

I also figured out how I could find the registry keys using the proper powershell command.  However it doesn't seem like that really provides a significant advantage, even if I moved the query into a type/provider and just put a registry entry here to avoid use of exec entirely.  Furthermore, I imagine microsoft is more likely to maintain the powershell format going forward than the registry keys.